vulnerabilities/pipelines/vmwarephoton_importer.py

```diff
@@ -0,0 +1,153 @@
+from univers.versions import GenericVersion  # Import GenericVersion
+
+from vulnerabilities import severity_systems
+from vulnerabilities.importer import AdvisoryData
+from vulnerabilities.importer import AffectedPackage
+from vulnerabilities.importer import Reference
+from vulnerabilities.importer import VulnerabilitySeverity
+from vulnerabilities.pipelines import VulnerableCodeBaseImporterPipeline
+from vulnerabilities.utils import get_cwe_id
+from vulnerabilities.utils import get_item
+
+
+class VMWAREPHOTONImporterPipeline(VulnerableCodeBaseImporterPipeline):
+    """Collect advisories from VMWARE_PHOTON."""
+
+    '''
+    EXAMPLE:
+    {
+    "cve_id": "CVE-2020-11979",
+    "pkg": "apache-ant",
+    "cve_score": 7.5,
+    "aff_ver": "all versions before 1.10.8-2.ph1 are vulnerable",
+    "res_ver": "1.10.8-2.ph1"
+    }
+    '''
+
+    pipeline_id = "vmwarephoton_importer"
+    repo_url = "https://github.com/vmware/photon/wiki/Security-Advisories"
+    spdx_license_expression = "CC BY-SA 4.0"
+    license_url = "https://creativecommons.org/licenses/by-sa/4.0/"
+
+    importer_name = "PHOTON Importer"
+
+    urls = [
+        "https://packages.vmware.com/photon/photon_cve_metadata/cve_data_photon1.0.json",
+        "https://packages.vmware.com/photon/photon_cve_metadata/cve_data_photon2.0.json",
+        "https://packages.vmware.com/photon/photon_cve_metadata/cve_data_photon3.0.json",
+        "https://packages.vmware.com/photon/photon_cve_metadata/cve_data_photon4.0.json",
+        "https://packages.vmware.com/photon/photon_cve_metadata/cve_data_photon5.0.json",
+    ]
+
+    def advisories_count(self):
+        advisory_count = 0
+        for url in self.urls:
+            try:
+                response = requests.get(url)
+                response.raise_for_status()
+                data = response.json()
+                advisory_count += len(data)
+            except requests.HTTPError as http_err:
+                self.log(
+                    f"HTTP error occurred while fetching {url}: {http_err} \n {traceback_format_exc()}",
+                    level=logging.ERROR,
+                )
+            except requests.RequestException as req_err:
+                self.log(
+                    f"Request exception occurred while fetching {url}: {req_err} \n {traceback_format_exc()}",
+                    level=logging.ERROR,
+                )
+            except Exception as e:
+                self.log(f"Unexpected error: {e} \n {traceback_format_exc()}", level=logging.ERROR)
+        return advisory_count
+
+    def collect_advisories(self) -> Iterable[AdvisoryData]:
+        advisories = []
+        for url in self.urls:
+            try:
+                response = requests.get(url)
+                response.raise_for_status()
+                advisories_data = response.json()  # Fetch the data from the API
+                advisories.extend(self.to_advisory(advisories_data))  # Collect advisories for each URL
```

Comment:
Not the right approach, advisory should always be yielded. No need to compute all advisory at once. See the other importer pipelines like https://github.com/aboutcode-org/vulnerablecode/blob/main/vulnerabilities/pipelines/nginx_importer.py